src/Nethermind/Nethermind.State/IWorldStateManager.cs

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Trie.Pruning;
+
+namespace Nethermind.State;
+
+public interface IWorldStateManager
+{
+    IWorldState GlobalWorldState { get; }
+    IStateReader GlobalStateReader { get; }
+
+    /// <summary>
+    /// Used by read only tasks that need to execute blocks.
+    /// Why does it need an `IStateReader`? I'm not sure, but I don't wanna break things.
+    /// The Action here is a resetter. Previously an explicit DbProvider's read only implementation need to be reset.
+    /// </summary>
+    /// <returns></returns>
+    (IWorldState, IStateReader, Action) CreateResettableWorldState();
```

Comment:
Why not just return one object that implements all of those interfaces?

Reply:
I'm just trying to not break things.

`IWorldState` have state while `IStateReader` does not, so they can't be combined. As for why even `IStateReader` is re-created. I've no idea.